code/go/pipelines-workshop/infra/workshop-stack.go

```diff
@@ -0,0 +1,74 @@
+package infra
+
+import (
+	"cdk-workshop/hitcounter"
+
+	"github.com/aws/aws-cdk-go/awscdk/v2"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awsapigateway"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awslambda"
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+	"github.com/cdklabs/cdk-dynamo-table-viewer-go/dynamotableviewer"
+)
+
+type CdkWorkshopStackProps struct {
+	awscdk.StackProps
+}
+
+type cdkWorkshopStack struct {
+	awscdk.Stack
```

Comment:
This isn't really the right place for this? I don't know if this is auto-generated or not but you don't typically place interface declarations inside a struct.

Reply:
This is actually what Romain suggested to me when I was struggling to figure out how to "extend" a "class". The way this works makes sense to me within my limited experience of the language, what would you suggest as an alternative?